macros/src/main/scala/pureconfig/Derivation.scala

```diff
@@ -1,11 +1,17 @@
 package pureconfig
 
+import scala.annotation.implicitNotFound
 import scala.collection.mutable
 import scala.language.experimental.macros
 import scala.reflect.macros.whitebox
 
 import pureconfig.derivation._
 
+@implicitNotFound("""Cannot find an implicit Derivation for ${A}.
+If your application does not define a Derivation elsewhere,
+consider using PureConfig's auto derivation by defining the following:
```

Comment:
What do you think about making the message describe _where_ the `import` needs to be added? A common problem seems to be that users fail to add it at the site of the `loadConfig*` call.

Reply:
so something like:
>"""Cannot find an implicit Derivation for ${A}.
If your application does not define a Derivation elsewhere,
consider using PureConfig's auto derivation by defining the following:
>
>import pureconfig.generic.auto._
>
>at the scope of your pureconfig.loadConfig[${A}] call"""
?

Reply:
Yeah, that's roughly what I was thinking. I suggest you wait until @ruippeixotog takes a look before making any changes.